src/app_service/service/activity_center/service.nim

```diff
@@ -0,0 +1,181 @@
+
+import eventemitter
+import ../../../app/core/[main]
+import ../../../app/core/tasks/[qt, threadpool]
+
+import web3/ethtypes, web3/conversions, stew/byteutils, nimcrypto, json_serialization, chronicles
+import json, tables, json_serialization
+
+import ../chat/service as chat_service
+
+import status/statusgo_backend_new/activity_center as status_activity_center
+import status/statusgo_backend_new/response_type
+import ./dto/notification
+
+export notification
+
+include async_tasks
+
+logScope:
+  topics = "activity-center-service"
+
+type
+  ActivityCenterNotificationsArgs* = ref object of Args
+    activityCenterNotifications*: seq[ActivityCenterNotificationDto]
+
+  MarkAsReadNotificationProperties* = ref object of Args
+    communityId*: string
+    channelId*: string
+    notificationTypes*: seq[ActivityCenterNotificationType]
+
+  MarkAsUnreadNotificationProperties* = ref object of Args
+    communityId*: string
+    channelId*: string
+    notificationTypes*: seq[ActivityCenterNotificationType]
+
+# Signals which may be emitted by this service:
+# TODO remove new when refactor is done
+const SIGNAL_ACTIVITY_CENTER_NOTIFICATIONS_LOADED* = "activityCenterNotificationsLoaded-new"
+const SIGNAL_MARK_NOTIFICATIONS_AS_READ* = "markNotificationsAsRead-new"
+const SIGNAL_MARK_NOTIFICATIONS_AS_UNREAD* = "markNotificationsAsUnread-new"
+
+proc getActivityCenterNotifications*(cursor: string = ""): (string, seq[ActivityCenterNotificationDto]) =
+  var cursorVal: JsonNode
+  
+  if cursor == "":
+    cursorVal = newJNull()
+  else:
+    cursorVal = newJString(cursor)
+
+  let callResult = status_activity_center.rpcActivityCenterNotifications(cursorVal, 20)
+  result = parseActivityCenterNotifications(callResult.result)
+
+
+QtObject:
+  type Service* = ref object of QObject
+    threadpool: ThreadPool
+    events: EventEmitter
+    chatService: chat_service.Service
+    cursor*: string
+
+  # Forward declaration
+  proc asyncActivityNotificationLoad*(self: Service)
+
+  proc delete*(self: Service) =
+    self.QObject.delete
+
+  proc newService*(
+      events: EventEmitter,
+      threadpool: ThreadPool,
+      chatService: chat_service.Service
+      ): Service =
+    new(result, delete)
+    result.QObject.setup
+    result.events = events
+    result.threadpool = threadpool
+    result.chatService = chatService
+
+  proc init*(self: Service) =
+    self.asyncActivityNotificationLoad()
+
+  proc hasMoreToShow*(self: Service): bool =
+    return self.cursor != ""
+  
+  proc asyncActivityNotificationLoad*(self: Service) =
+    let arg = AsyncActivityNotificationLoadTaskArg(
+      tptr: cast[ByteAddress](asyncActivityNotificationLoadTask),
+      vptr: cast[ByteAddress](self.vptr),
+      slot: "asyncActivityNotificationLoaded"
+    )
+    self.threadpool.start(arg)
+
+  proc getActivityCenterNotifications*(self: Service, initialLoad: bool = true): seq[ActivityCenterNotificationDto] =
```

Comment:
>  since we know about that based on the value of the self.cursor variable (if it's empty it's about initial loading, if it's not, then it's not initial loading)

That's not totally true. If we already did the initial load and the cursor was returned as `""`, that means that there are no other notifications to fetch.
that is why we have the codition on line 89: 
```nim
    # Notifications were already loaded, since cursor will 
    # be nil/empty if there are no more notifs
    if(not initialLoad and self.cursor == ""): return
    ```

Reply:
@jrainville yes, you're right, if cursor was returned as `""`, that means that there are no other notifications to fetch.

But I am interested more in `initialLoad` param, do we need it at all as a parm in this context if we already have the mechanism to fetch notifications in an async way as we did that in the `init` method of this service? Why do we need this `getActivityCenterNotifications` method at all?